python/grass/jupyter/interactivemap.py

```diff
@@ -347,13 +347,59 @@ def add_layer_control(self, **kwargs):
         else:
             self.layer_control_object = self._ipyleaflet.LayersControl(**kwargs)
 
+    def add_features(self):
+        """Add custom features like query button and coordinate retrieval"""
+        import ipywidgets as widgets
+        from ipyleaflet import WidgetControl
+
+        # A button to activate/deactivate query mode
+        query_button = widgets.Button(description="Activate Query Mode")
+
+        # Variable to store the state of query mode
+        self.query_mode = False
+
+        # Output widget to display query results
+        output_widget = widgets.Output(
+            layout={"border": "1px solid black", "display": "none"}
+        )
+
+        # Function to toggle query mode
+        def toggle_query_mode(button):
+            self.query_mode = not self.query_mode
+            button.description = (
+                "Deactivate Query Mode" if self.query_mode else "Activate Query Mode"
+            )
+            output_widget.layout.display = "block" if self.query_mode else "none"
+
+        query_button.on_click(toggle_query_mode)
+
+        # Add the button to the map using WidgetControl
+        query_control = WidgetControl(widget=query_button, position="topright")
+        self.map.add_control(query_control)
+
+        # Create an output widget to display query results
+        output_control = WidgetControl(widget=output_widget, position="bottomright")
+        self.map.add_control(output_control)
+
+        # Function to handle map click for querying
+        def handle_interaction(**kwargs):
+            if self.query_mode and kwargs.get("type") == "click":
+                latlon = kwargs.get("coordinates")
+                # lat, lon = latlon[0], latlon[1]
```

Comment:
Here you call the m.proj to transform the coordinates and use those coordinates to call d.rast/d.vect